misk/src/test/kotlin/misk/client/TypedHttpClientInterceptorTest.kt

```diff
@@ -0,0 +1,177 @@
+package misk.client
+
+import com.google.inject.Guice
+import com.google.inject.Provides
+import com.google.inject.util.Modules
+import helpers.protos.Dinosaur
+import misk.Action
+import misk.MiskModule
+import misk.NetworkChain
+import misk.NetworkInterceptor
+import misk.inject.KAbstractModule
+import misk.inject.addMultibinderBinding
+import misk.inject.getInstance
+import misk.inject.to
+import misk.moshi.MoshiModule
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import misk.testing.TestWebModule
+import misk.web.Post
+import misk.web.RequestBody
+import misk.web.RequestContentType
+import misk.web.ResponseContentType
+import misk.web.WebActionModule
+import misk.web.WebModule
+import misk.web.actions.WebAction
+import misk.web.jetty.JettyService
+import misk.web.mediatype.MediaTypes
+import okhttp3.Response
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.http.Body
+import retrofit2.http.POST
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@MiskTest(startService = true)
+internal class TypedHttpClientInterceptorTest {
+  @MiskTestModule
+  val module = Modules.combine(
+      MiskModule(),
+      WebModule(),
+      TestWebModule(),
+      TestModule()
+  )
+
+  @Inject
+  private lateinit var jetty: JettyService
+
+  private lateinit var client: ReturnADinosaur
+
+  @BeforeEach
+  fun createClient() {
+    val clientInjector = Guice.createInjector(MoshiModule(), ClientModule(jetty))
+    client = clientInjector.getInstance()
+  }
+
+  @Test
+  fun useTypedClient() {
```

Comment:
Maybe a test for if an Interceptor throws? E.g. a ClientApplicationInterceptor that checks that you have a certain header in your request?